test/tap/version-from-git.js

```diff
@@ -0,0 +1,162 @@
+var common = require('../common-tap.js')
+var fs = require('fs')
+var path = require('path')
+
+var mkdirp = require('mkdirp')
+var osenv = require('osenv')
+var rimraf = require('rimraf')
+var test = require('tap').test
+
+var npm = require('../../lib/npm.js')
+
+var pkg = path.resolve(__dirname, 'version-from-git')
+var packagePath = path.resolve(pkg, 'package.json')
+var cache = path.resolve(pkg, 'cache')
+
+var json = { name: 'cat', version: '0.1.2' }
+
+test('npm version from-git with a valid tag creates new commit', function (t) {
+  var version = '1.2.3'
+  var tag = 'v' + version
+  setup()
+  createTag(t, tag, runVersion)
+
+  function runVersion (er) {
+    t.ifError(er, 'git tag ran without error')
+    npm.config.set('sign-git-tag', false)
+    npm.commands.version(['from-git'], checkVersion)
+  }
+
+  function checkVersion (er) {
+    var git = require('../../lib/utils/git.js')
+    t.ifError(er, 'version command ran without error')
+    git.whichAndExec(
+      ['log'],
+      { cwd: pkg, env: process.env },
+      checkCommit
+    )
+  }
+
+  function checkCommit (er, log, stderr) {
+    t.ifError(er, 'git log ran without issue')
+    t.notOk(stderr, 'no error output')
+    t.ok(log.indexOf(version) !== -1, 'commited from subdirectory')
+    t.end()
+  }
+})
+
+test('npm version from-git with a valid tag updates the package.json version', function (t) {
+  var version = '1.2.3'
+  var tag = 'v' + version
```

Comment:
This should use `npm.config.get('tag-version-prefix')` instead of hardcoding the `v`, in order to preserve the symmetry of that configuration. (Also there should be a test case for using a `tag-version-prefix` other than `v`.)